mlem/core/dataset_type.py

```diff
@@ -193,6 +252,40 @@ def get_model(self) -> Type[BaseModel]:
         )
 
 
+class ListWriter(DatasetWriter):
+    type: ClassVar[str] = "list"
+
+    def write(
+        self, dataset: DatasetType, storage: Storage, path: str
+    ) -> Tuple[DatasetReader, Dict]:
+        assert isinstance(dataset, ListDatasetType)
+        res = {}
+        readers = []
+        for i in range(dataset.size):
+            elem = dataset.data[i]
+            elem_reader, art = dataset.dtype.get_writer().write(
+                DatasetType.create(elem), storage, posixpath.join(path, str(i))
```

Comment:
Running `DatasetAnalyzer` is slow, so let's get rid of it. You already have underlying DatasetType so use that like `dataset.dtype.copy().bind(elem)`